nncf/experimental/tensorflow/graph/argprovider.py

```diff
@@ -0,0 +1,229 @@
+        check_port_id(input_port_id, min_port_id=0, max_port_id=1)
+
+        if input_port_id == 0:
+            return args[0]
+
+        return kwargs['filter']  # input_port_id == 1
+
+    def set_input(self, input_port_id: int, value, args, kwargs):
+        check_port_id(input_port_id, min_port_id=0, max_port_id=1)
+
+        if input_port_id == 0:
+            return replace(args, input_port_id, value), kwargs
+
+        kwargs['filter'] = value
+        return args, kwargs
+
+    def get_output(self, output_port_id: int, args, kwargs):
+        check_port_id(output_port_id, min_port_id=0, max_port_id=0)
+
+        if len(args) > 1:
+            raise ValueError(f'Unexpected `args`: {args}')
+
+        return args[output_port_id]
+
+    def set_output(self, output_port_id: int, value, args, kwargs):
+        check_port_id(output_port_id, min_port_id=0, max_port_id=0)
+
+        if len(args) > 1:
+            raise ValueError(f'Unexpected `args`: {args}')
+
+        return replace(args, output_port_id, value), kwargs
+
+
+@TF_ARG_PROVIDERS.register('FusedBatchNormV3')
+class FusedBatchNormV3ArgProvider(ArgProvider):
+    """
+    Argument provider of the `FusedBatchNormV3` operation.
+    Outputs:
+        port_id: 0 - output tensor (`y` key).
+    """
+
+    def get_output(self, output_port_id: int, args, kwargs):
+        check_port_id(output_port_id, min_port_id=0, max_port_id=0)
+        return args[0].y
+
+    def set_output(self, output_port_id: int, value, args, kwargs):
+        check_port_id(output_port_id, min_port_id=0, max_port_id=0)
+        x = args[0]._replace(y=value)
+        return replace(args, 0, x), kwargs
+
+
+@TF_ARG_PROVIDERS.register('DepthwiseConv2dNative')
+class DepthwiseConv2dNativeArgProvider(ArgProvider):
+    """
+    Argument provider of the `DepthwiseConv2dNative` operation.
+    Inputs:
+        port_id: 0 - input tensor (args[0]).
+        port_id: 1 - filter tensor (args[1]).
+    """
+
+    def get_input(self, input_port_id: int, args, kwargs):
+        check_port_id(input_port_id, min_port_id=0, max_port_id=1)
+        return args[input_port_id]
+
+    def set_input(self, input_port_id: int, value, args, kwargs):
+        check_port_id(input_port_id, min_port_id=0, max_port_id=1)
+        return replace(args, input_port_id, value), kwargs
+
+
+@TF_ARG_PROVIDERS.register('MatMul')
+class MatMulArgProvider(ArgProvider):
+    """
+    Argument provider of the `MatMul` operation.
+    Inputs:
+        port_id: 0 - input tensor (args[0]).
+        port_id: 1 - filter (always?) tensor (args[1]).
+    """
+
+    def get_input(self, input_port_id: int, args, kwargs):
+        check_port_id(input_port_id, min_port_id=0, max_port_id=1)
+        return args[input_port_id]
+
+    def set_input(self, input_port_id: int, value, args, kwargs):
+        check_port_id(input_port_id, min_port_id=0, max_port_id=1)
+        return replace(args, input_port_id, value), kwargs
```

Comment:
How do we perform develop-time checks that all ops that should have arg providers defined actually have them defined?

Reply:
It cannot be understood at the moment.